content/3_Silver/Prefix_Sums_2.mdx

```diff
@@ -439,6 +439,108 @@ int main() {
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+ 
+public class ForestQueries3 {
+    public static void main(String[] args) throws IOException {
+        Reader r = new Reader();
+        PrintWriter pw = new PrintWriter(System.out);
+ 
+        int n = r.nextInt();
+        int q = r.nextInt();
+        int N = n + 1;
+        int[][] forest = new int[N][N];
+        for (int i = 1; i < N; i++) {
+            for (int j = 1; j < N; j++) {
+                if (r.isTree()) {
+                    forest[i][j] = 1;
+                }
+            }
+        }
+ 
+        for (int i = 1; i < N; i++) {
+            for (int j = 1; j < N; j++) {
+                forest[i][j] += forest[i][j - 1];
+            }
+        }
+        for (int j = 1; j < N; j++) {
+            for (int i = 1; i < N; i++) {
+                forest[i][j] += forest[i - 1][j];
+            }
+        }
+        StringBuilder sb = new StringBuilder();
+        while (q-- > 0) {
+            int y1 = r.nextInt();
+            int x1 = r.nextInt();
+            int y2 = r.nextInt();
+            int x2 = r.nextInt();
+            sb.append(forest[y2][x2] - forest[y1 - 1][x2] - forest[y2][x1 - 1] + forest[y1 - 1][x1 - 1]).append("\n");
```

Comment:
over 80 characters?

Reply:
https://usaco.guide/general/contributing#code-conventions

Reply:
also use tabs instead of spaces to indent

Reply:
My editor is set up such that it auto replaces tabs with spaces. If you want, I can disable this or just find and replace spaces with the tab literal.